auth.py

```
@@ -0,0 +1,323 @@
+    if not row:
+        return None
+
+    # Session versioning: if user bumped their version, invalidate
+    if row["sess_ver"] != row["user_ver"]:
+        return None
+
+    return dict(row)
+
+
+async def require_auth(
+    request: Request, pool: asyncpg.Pool = Depends(get_pool)
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)